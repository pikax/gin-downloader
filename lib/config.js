/**
 * Created by rodriguesc on 03/03/2017.
 */
"use strict";