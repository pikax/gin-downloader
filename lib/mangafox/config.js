'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
/**
 * Created by rodriguesc on 03/03/2017.
 */

var debug = require('debug')('gin-downloader:mangafox:config');

var site = 'http://mangafox.me';
var config = {
  name: 'MangaFox',
  site: site,
  mangas_url: site + '/manga',
  latest_url: site + '/releases'
};

exports.default = config;

debug('using %O', config);