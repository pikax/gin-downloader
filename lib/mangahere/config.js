'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
/**
 * Created by rodriguesc on 03/03/2017.
 */

var debug = require('debug')('gin-downloader:mangahere:config');

var site = 'http://www.mangahere.co';
var config = {
  name: 'MangaHere',
  site: site,
  mangas_url: site + '/mangalist',
  latest_url: site + '/latest'
};

exports.default = config;

debug('using %O', config);