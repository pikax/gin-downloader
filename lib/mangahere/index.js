'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _lodash = require('lodash');

var _lodash2 = _interopRequireDefault(_lodash);

var _parser = require('./parser');

var _parser2 = _interopRequireDefault(_parser);

var _config = require('./config');

var _config2 = _interopRequireDefault(_config);

var _request = require('../common/request');

var _names = require('./names');

var _helper = require('../common/helper');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/**
 * Created by rodriguesc on 03/03/2017.
 */
var debug = require('debug')('gin-downloader:mangahere');
var verbose = require('debug')('gin-downloader:mangahere:verbose');

var mangas = function mangas() {
  debug('getting mangas');
  return (0, _helper.getDoc)(_config2.default.mangas_url).then(_parser2.default.mangas).tap(function (x) {
    return debug('mangas: ' + x.length);
  });
};

var latest = function latest() {
  debug('getting latest');
  return (0, _helper.getDoc)(_config2.default.latest_url).then(_parser2.default.latest).tap(function (x) {
    return debug('found: ' + x.length);
  });
};

var info = function info(name) {
  debug('getting info for ' + name);

  var src = (0, _names.resolveUrl)(name);

  debug('getting info ' + src);
  return (0, _helper.getDoc)(src).then(_parser2.default.mangaInfo).tap(function (x) {
    return debug('info: %s', x);
  });
};

var chapters = function chapters(name) {
  debug('getting chapters ' + name);
  var src = (0, _names.resolveUrl)(name);

  return (0, _helper.getDoc)(src).then(_parser2.default.chapters).tap(function (x) {
    if (!x || x.length === 0) throw 'not found';
  }).catch(function (x) {
    throw new Error('Chapters not found', x);
  }).tap(function (x) {
    return debug('chapters: ' + x.length);
  });
};

var images = function images(name, chapter) {
  debug('getting images ' + name + ':' + chapter);

  return chapters(name).then(function (chaps) {
    return _lodash2.default.find(chaps, { number: name + ' ' + chapter });
  }).then(function (x) {
    if (!x) throw new Error('Manga: ' + name + ' chapter ' + chapter + ' doesn\'t exists.');
    return x.src;
  }).then(imagesByUrl);
};

var imagesByUrl = function imagesByUrl(uri) {
  debug('getting images for ' + uri);

  return imagesPaths(uri).tap(function (x) {
    return debug('images: ' + x.length);
  }).then(function (x) {
    return x.map(image);
  }).tap(function (x) {
    return debug('images resolved : ' + x);
  });
};

var imagesPaths = function imagesPaths(uri) {
  debug('getting image paths ' + uri);
  return (0, _helper.getDoc)(uri).then(_parser2.default.imagesPaths).tap(function (x) {
    return debug('found ' + x.length + ' images:\n' + x);
  });
};

var image = function image(uri) {

  debug('getting image from ' + uri);

  return (0, _request.getHtml)(uri).tap(function (x) {
    return verbose('html: %s', x);
  }).then(_parser2.default.image);
};

var resolve = function resolve(name, chapter) {
  debug('resolve ' + name + ':' + chapter);

  return images(name, chapter).tap(function (x) {
    return debug('found %o', x);
  });
};

exports.default = {
  config: _config2.default,

  mangas: mangas,
  info: info,
  chapters: chapters,
  images: images,
  imagesPaths: imagesPaths,
  resolve: resolve,
  latest: latest
};