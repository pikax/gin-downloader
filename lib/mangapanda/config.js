/**
 * Created by rodriguesc on 05/03/2017.
 */

const site= "http://www.mangapanda.com";

export  default  {
    name : "MangaPanda",
    site : site,
    mangas_url : site+"/alphabetical",
    latest_url : site+"/latest"
};