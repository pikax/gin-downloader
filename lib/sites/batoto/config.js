"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var url_1 = require("url");
var site = "http://bato.to/";
exports.config = {
    name: "Batoto",
    site: site,
    mangas_url: url_1.resolve(site, "/search_ajax"),
    latest_url: url_1.resolve(site, "/"),
};
exports.default = exports.config;
exports.debug = require("debug")("gin-downloader:batoto");
exports.verbose = require("debug")("gin-downloader:batoto:verbose");
exports.verbose("using %O", exports.config);
//# sourceMappingURL=config.js.map