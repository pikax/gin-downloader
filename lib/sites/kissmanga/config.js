"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var url_1 = require("url");
var site = "http://kissmanga.com/";
exports.config = {
    name: "KissManga",
    site: site,
    mangas_url: url_1.resolve(site, "/MangaList/"),
    latest_url: url_1.resolve(site, "/MangaList/LatestUpdate")
};
exports.default = exports.config;
exports.debug = require("debug")("gin-downloader:kissmanga");
exports.verbose = require("debug")("gin-downloader:kissmanga:verbose");
exports.verbose("using %O", exports.config);
//# sourceMappingURL=config.js.map