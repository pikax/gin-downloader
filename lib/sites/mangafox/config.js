"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var url_1 = require("url");
var site = "http://mangafox.me";
exports.config = {
    name: "MangaFox",
    site: site,
    mangas_url: url_1.resolve(site, "/manga/"),
    latest_url: url_1.resolve(site, "/releases/")
};
exports.default = exports.config;
exports.debug = require("debug")("gin-downloader:mangafox");
exports.verbose = require("debug")("gin-downloader:mangafox:verbose");
exports.verbose("using %O", exports.config);
//# sourceMappingURL=config.js.map