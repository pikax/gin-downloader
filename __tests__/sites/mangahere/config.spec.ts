import {MangaHereConfig} from "../../../src/manga/mangahere/config";

describe("MangaHere config", () => {

    it("should have correct values", () => {
        const config = new MangaHereConfig();

        expect(config).toMatchSnapshot();


        // NOTE config is only readOnly, Jest doesn't resolve readOnly props
        const e = {
            name: config.name,
            latestUrl: config.latestUrl,
            mangasUrl: config.mangasUrl,
            site: config.site,
        };

        expect(e).toMatchSnapshot();
    });

});


