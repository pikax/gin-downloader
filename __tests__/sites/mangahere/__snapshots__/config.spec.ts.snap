// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MangaHere config should have correct values 1`] = `MangaHereConfig {}`;

exports[`MangaHere config should have correct values 2`] = `
Object {
  "latestUrl": "http://www.mangahere.cc/latest/",
  "mangasUrl": "http://www.mangahere.cc/mangalist/",
  "name": "MangaHere",
  "site": "http://www.mangahere.cc",
}
`;
